### PR TITLE
fix

### PR DESCRIPTION
**Pull request by miclegoden**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
 import logo from './logo.svg'
-import TrafficLight from './Compoenet/TrafficLight'
+import TrafficLight from './Component/TrafficLight'
 import './App.css'
 
+const timeDuration = [1000, 2000, 4000]
+const colors = ['Red', 'Yellow', 'Green']
+
 function App() {
+  const [firstLight, setFirstLight] = useState<number>(0)
```
Comment (hungry-engineer) [inline src/App.tsx:10]: firstLightColorIndex make much more sense.It's actually index, not color. Right?

File: src/App.tsx [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      setFirstLight((firstLight) => (firstLight + 1) % 3)
+    }, timeDuration[firstLight])
+
+    return () => clearTimeout(timeout)
+  }, [firstLight])
```
Comment (hungry-engineer) [inline src/App.tsx:18]: In terms of timer interval, there might be some slight differences. I mean it's not accurately 1,2, 4s maybe.

File: src/Component/Light.tsx [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
     style={{ backgroundColor: props.color, opacity: props.active ? 1 : 0.4 }}
   />
 )
 
-export declare interface TrafficLightProps {
-  color: string
-  active: boolean
-}
+// export declare interface TrafficLightProps {
+//   color: string
```
Comment (hungry-engineer) [inline src/Component/Light.tsx:16]: Is this needed? Please remove code if not needed rather than disabling

File: src/Component/TrafficLight.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+// const lightDurations = [1000, 2000, 4000]
+
+type TrafficLightProps = {
+  lightActive: string
+}
+
+const colors = ['Red', 'Yellow', 'Green']
+
+const TrafficLight = (props: TrafficLightProps) => {
```
Comment (hungry-engineer) [inline src/Component/TrafficLight.tsx:13]: TrafficLight and Light? LightBox would make much more sense.

File: src/Component/Light.tsx [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
+//   active: boolean
+// }
```
Comment (hungry-engineer) [inline src/Component/Light.tsx:18]: Same here,Remove unncessary code.

**Comment by miclegoden** [comment]
Thank you so much

**Pull request merged**